### PR TITLE
ARM Debugging

### PR DESCRIPTION
**Pull request by smasher816**
Everything is placed in a separate arm_cortex file so that other chip families (kinetis, nrf52, future chips) can also benefit.

May tracking down Hardfaults and unused_isr's forever be less of a PITA

### CODE REVIEW

File: Lib/arm_cortex.c [hunk 1]
```diff
@@ -0,0 +1,91 @@
+#include <stdint.h>
```
Comment (haata) [inline Lib/arm_cortex.c]: Add an MIT header, see https://github.com/kiibohd/controller/blob/master/Lib/nrf5.c#L1

File: Lib/arm_cortex.h [hunk 2]
```diff
@@ -0,0 +1,132 @@
+#pragma once
```
Comment (haata) [inline Lib/arm_cortex.h]: Again, add a header: https://github.com/kiibohd/controller/blob/master/Lib/nrf5.c#L1

File: Lib/arm_cortex.c (hunk 1 above)
Comment (smasher816) [reply Lib/arm_cortex.c]: Done

File: Lib/arm_cortex.h (hunk 2 above)
Comment (smasher816) [reply Lib/arm_cortex.h]: Done

**Pull request merged**